### DISCUSSION TITLE
Suggest report 「 建议反馈」About dde on arch linux

### DISCUSSION THREAD

**Discussion by rensilin** [linuxdeepin/developer-center#2833] [General | 常规讨论]
### main 发这条issue的主要目标

1. 询问官方如何看待arch上的dde。Ask the official how do you view the dde on arch.
2. 是否会关注github上的PR。Will You pay attention to the PR on github.

### background 背景

我从deepin14开始接触linux，对我来说，deepin确实是一个漂亮、实用而且易用(虽然有些小BUG)的操作系统。切换到arch是因为在deepin上搭建开发环境确实是一件很困难的事情，可能因为发版模式的问题，当时软件仓库的版本都比较老，常遇见各种冲突。

最近我组装了新电脑，装系统时发现了arch上也支持了dde，两件快乐的事情重合在一起。而这两份快乐，又给我带来更多的快乐。得到的，本该是像梦境一般的开发体验……但是，为什么，会变成这样呢。

几个严重Bug困扰了我几个月时间：

1. 开机强制卡30秒。见linuxdeepin/developer-center#2490。
4. 桌面背景无法切换，设置部分页面打不开。
5. 最严重的是两周前更新后，桌面直接打不开了。见https://github.com/linuxdeepin/developer-center/issues/2251。

我相信就算开机卡30秒可以忍受，但是如果无法使用对于任何一个软件来说都是无法容忍的问题。在发现已经有其他人提出issue，但是官方因为没有arch的环境，缺少现场日志所以无法解决问题的时候，我尝试自己能否解决。好在dde的项目都比较好入手，也没花特别多时间，代码改动也不大，最后修好了启动超时和桌面的问题。看了Developer Guide后，提交了PR\[[1](https://github.com/linuxdeepin/startdde/pull/101), [2](https://github.com/linuxdeepin/dde-file-manager/pull/426)\]，PR里也描述了问题和解决方案。

我自己编译后能解决我的系统问题，但是提交的PR好像也没有后续了，如果官方不合入，其他使用者也就只能暂时抛弃dde。而且最近几次因为更新了qt等其他库，即使是保持dde不更新，我的系统也经常打不开，只能每次更新后都重新编译。

## Other：commit guide is invisiable

> 若是提交贡献 Pull requests, 请注意commit 提交规范, 参见 :  [Commit提交规范](https://github.com/linuxdeepin/developer-center/wiki/Commit%E6%8F%90%E4%BA%A4%E8%A7%84%E8%8C%83)

这个规范打不开

**Comment by BLumia** [comment]
> 如何看待arch上的dde

Arch 上的 DDE 包主要由 @felixonmars 和 @justforlxz （AUR） 进行打包（虽然官方支持的发行版大概只有 Deepin/UOS，因为主要的公司内开发人员是用 Deepin/UOS 的...）。

> 是否会关注github上的PR

tl;dr. 之前确实关注很少并且出于一些原因没法直接合并外部的 PR，不过最近的阶段会重新规划这件事。详见下面一小段。

> 我自己编译后能解决我的系统问题，但是提交的PR好像也没有后续了，如果官方不合入，其他使用者也就只能暂时抛弃dde。

Deepin 的开发流程其实之前蛮 Company-driven 的，不过在逐渐的试着变得更加开放并拥抱社区。之前相当一段时间（从 v20 刚开始那段时间起）因为公司内部开发是在公司内网的 git 仓库进行的，导致很难直接合并来自 GitHub 的 patch，这个问题持续到了现在。当然我们最近（就是这段时间）在逐步把我们的仓库和社区开发流程迁回 GitHub 来解决与社区开发有脱节的问题，迁移完毕后相应的 patch 应该会能更方便的合并进来。

> 看了Developer Guide后，提交了PR https://github.com/linuxdeepin/startdde/pull/101, https://github.com/linuxdeepin/dde-file-manager/pull/426 ，PR里也描述了问题和解决方案。

这两个仓库的开发流程还没迁移回 GitHub ，所以**暂时**大概不能直接合 PR，我这边记录下然后帮你跟进下情况~

> 这个规范打不开

抱歉，因为 wiki 很久未更新，最近更新了一部分过时的 wiki 页面，没注意到别的地方引用到 wiki 页面的地方，随后会更新相关的链接。提交规范见 [Commit 提交信息规范](https://github.com/linuxdeepin/developer-center/wiki/Commit-%E6%8F%90%E4%BA%A4%E4%BF%A1%E6%81%AF%E8%A7%84%E8%8C%83)。若您发现有其它地方内容过时/链接失效的情况也欢迎反馈。

**Comment by justforlxz** [comment]
最近在集中修复arch dde的问题

linuxdeepin/developer-center#2490 这个问题已经在 https://github.com/linuxdeepin/startdde/commit/68b8f01d10842035551062858c28b643ce3ab0e2 修复了。

现在正在修复使用非 deepin-kwin 时的问题，会尽快解决。

**Comment by rensilin** [comment]
感谢解答，理解这些工作的困难。发这个也是因为滚挂了3次了，恶性BUG的优先级应该可以提高下，也不是必须要直接合我的PR，只要能解决问题就行。
期待dde变得更好。

**Comment by hillwoodroc** [comment]
在 openSUSE 上倒是不卡， 只是进入桌面有点慢，要过一段时间 dock 才显示出来。

**Comment by zccrs** [comment]
> 在 openSUSE 上倒是不卡， 只是进入桌面有点慢，要过一段时间 dock 才显示出来。

不仅是arch，非deepin上DDE我们都会一起处理相关的问题，很多问题应该都是类似的。
